app/queries/get_link_changes.rb

```diff
@@ -0,0 +1,79 @@
+module Queries
+  class GetLinkChanges
+    # Current maximum number of results. This is an arbitrary number to prevent
+    # too much data being returned, and to hold off from implementing pagination.
+    MAXIMUM_NUMBER_OF_RESULTS = 250
+
+    attr_reader :params
+
+    def initialize(params)
+      @params = params
+    end
+
+    def as_json
+      results = link_changes.map do |link_change|
+        {
+          id: link_change.id,
+          source: expand_edition(link_change.source_content_id),
+          target: expand_edition(link_change.target_content_id),
+          link_type: link_change.link_type,
+          change: link_change.change,
+          user_uid: link_change.action.user_uid,
+          created_at: link_change.created_at
+        }
+      end
+
+      {
+        link_changes: results,
+      }
+    end
+
+  private
+
+    def expand_edition(content_id)
+      all_related_editions
+        .find { |edition| edition.content_id == content_id }
```

Comment:
I don't think you have any guarantee that this will return something. 

You might do better grouping editions by content_id to avoid iterating?

Reply:
I think Tijmen or Jos have changed the implementation now.